### PR TITLE
fixes #37: Feature Prisma ORM integration

### LINKED ISSUE #37: [Feat]: Use Prisma instead of Raw mongodb
### Is your proposal related to a problem?

Using Raw mongodb is not that efficient so,
Do you want to migrate your project to use `prisma` ORM it can be better and scalable.
I can migrate it. If you assign this issue to me.

### Describe the solution you'd like

I will use Prisma ORM and migrate everything to that including authentication

### Describe alternatives you've considered

It is going to be alternative of Raw mongodb schemas and a connection function that is not efficient in long run.

### Additional context

It can be very easy to use Prisma and creating new model is also easy with typesafety

### PR DESCRIPTION
**Pull request by PriyanshuPz**
- Fixes #37
- Adds Prisma ORM
- migration of all API routes
- **ENV Update**
- Uses in Prisma

    ```env
    DATABASE_URL=
    ```

- Command to run after first setup
```bash
npx prisma db push
```

### CODE REVIEW

**Comment by PriyanshuPz** [comment]
Hi @MitulSonagara,
Please check out the changes and merge the PR.

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

This changes having some issues with deployment

**Comment by PriyanshuPz** [comment]
@MitulSonagara 
I have fixed the error issue and verified by building it locally.
Please review the PR again and if every thing is working then merge it.

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

is there any changes with environment variables??

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

image

i am still having this error. I haven't used prisma before so you have to tell me process or changes during the deployment

**Comment by PriyanshuPz** [comment]
Oh! Ok @MitulSonagara 
in build command add 
```
npx prisma generate && next build
```

and add env
```env
DATABASE_URL
```

**Comment by PriyanshuPz** [comment]
@MitulSonagara Let me know if any other error.
I am active

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

only one little error

image

this is working correctly even if it says error as expected string

File: src/app/api/check-username-unique/route.ts [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
+
         if (!result.success) {
             const usernameErrors = result.error.format().username?._errors;
             return new Response(JSON.stringify({
                 success: false,
                 message: `${usernameErrors}`
             }), { status: 400 });
         }
-        
+
```
Comment (MitulSonagara) [inline src/app/api/check-username-unique/route.ts:31]: I think here some error occurred while checking unique username

Comment (PriyanshuPz) [reply src/app/api/check-username-unique/route.ts:31]: I fixed that by added 'force-dynamic'

**Comment by PriyanshuPz** [comment]
@MitulSonagara Please check again

File: src/app/api/check-username-unique/route.ts [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
         const result = UsernameQuerySchema.safeParse(queryParams);
-        
+
         if (!result.success) {
             const usernameErrors = result.error.format().username?._errors;
             return new Response(JSON.stringify({
                 success: false,
                 message: `${usernameErrors}`
             }), { status: 400 });
```
Comment (MitulSonagara) [inline src/app/api/check-username-unique/route.ts]: Make two changes for me 

Make status 200 here

Comment (MitulSonagara) [reply src/app/api/check-username-unique/route.ts]: image

and also here

**Review by MitulSonagara — CHANGES_REQUESTED** [review]

**Comment by PriyanshuPz** [comment]
@MitulSonagara See again PA,
Changes are done. Let me know if any other

**Review by MitulSonagara — APPROVED** [review]

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

PERFECT!!!

**Comment by MitulSonagara** [comment]
@priyanshuverma-dev 

Nice working with you...

**Pull request merged**